### PR TITLE
Update System.CommandLine version

### LINKED ISSUE #18952: Update CLI to stop using obsolete System.CommandLine APIs
See https://github.com/dotnet/sdk/pull/18922 for context

### PR DESCRIPTION
**Pull request by sfoslund**
Fixes https://github.com/dotnet/sdk/issues/18952

### CODE REVIEW

File: src/BuiltInTools/dotnet-watch/Program.cs [hunk 1]
```diff
@@ -153,7 +152,7 @@ internal static RootCommand CreateRootCommand(Func<CommandLineOptions, Task<int>
                 if (string.IsNullOrEmpty(options.Project))
                 {
 #pragma warning disable CS0618 // Type or member is obsolete
-                    var projectOptionShort = parseResults.ValueForOption<string>("-p");
+                    var projectOptionShort = parseResults.GetValueForOption(projectOption);
```
Comment (jonsequitur) [inline src/BuiltInTools/dotnet-watch/Program.cs:155]: You can remove the pragmas.

File: src/Cli/dotnet/commands/dotnet-build/BuildCommandParser.cs [hunk 2]
```diff
@@ -29,36 +34,45 @@ internal static class BuildCommandParser
         public static readonly Option<bool> NoLogoOption = new ForwardedOption<bool>("--nologo", LocalizableStrings.CmdNoLogo)
             .ForwardAs("-nologo");
 
-        public static readonly Option<bool> NoRestoreOption = CommonOptions.NoRestoreOption();
+        public static readonly Option<bool> NoRestoreOption = CommonOptions.NoRestoreOption;
 
-        public static readonly Option SelfContainedOption = CommonOptions.SelfContainedOption();
+        public static readonly Option SelfContainedOption = CommonOptions.SelfContainedOption;
```
Comment (jonsequitur) [inline src/Cli/dotnet/commands/dotnet-build/BuildCommandParser.cs:39]: Is an `Option<T>` an option here?

File: src/Cli/dotnet/commands/dotnet-format/common/FormatCommandCommon.cs [hunk 3]
```diff
@@ -71,21 +71,21 @@ public static void AddCommonDotnetFormatArgs(this List<string> dotnetFormatArgs,
...
-                parseResult.ValueForOption(VerbosityOption) is string { Length: > 0 } verbosity)
+                parseResult.GetValueForOption(VerbosityOption) is string { Length: > 0 } verbosity)
             {
                 dotnetFormatArgs.Add("--verbosity");
                 dotnetFormatArgs.Add(verbosity);
             }
 
             if (parseResult.HasOption(IncludeOption) &&
-                parseResult.ValueForOption(IncludeOption) is string[] { Length: > 0 } fileToInclude)
+                parseResult.GetValueForOption(IncludeOption) is string[] { Length: > 0 } fileToInclude)
```
Comment (jonsequitur) [inline src/Cli/dotnet/commands/dotnet-format/common/FormatCommandCommon.cs:81]: For all of these checks, is the `HasOption` check needed?

**Review by dsplaisted — APPROVED** [review]

**Review by joeloff — APPROVED** [review]
Looks good. Left some comments / questions.

**Comment by sfoslund** [comment]
@jonsequitur do you want to review the System.CommandLine updates before they're merged into 7.0.1xx?

**Review by joeloff — APPROVED** [review]

**Comment by JoeRobich** [comment]
@sfoslund Is it expected that this version of the command-line-api hasn't been published to the .NET 6 channel? How should other tools that ship in the SDK stay in sync?

**Comment by sfoslund** [comment]
We plan to update to the stable 2.0 version as soon as it's released (sometime in early December) so at that point I don't think we'll need to have these frequent updates anymore.

**Pull request merged**